### PR TITLE
Use logback instead of log4j to fix logging issues

### PR DESCRIPTION
**Pull request by jalaziz**
Another attempt at #226
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/jmxtrans/jmxtrans/pull/288%23issuecomment-100439696%22%2C%20%22https%3A//github.com/jmxtrans/jmxtrans/pull/288%23discussion_r29990731%22%2C%20%22https%3A//github.com/jmxtrans/jmxtrans/pull/288%23issuecomment-120799811%22%2C%20%22https%3A//github.com/jmxtrans/jmxtrans/pull/288%23issuecomment-120899379%22%2C%20%22https%3A//github.com/jmxtrans/jmxtrans/pull/288%23issuecomment-147812648%22%2C%20%22https%3A//github.com/jmxtrans/jmxtrans/pull/288%23issuecomment-147841391%22%2C%20%22https%3A//github.com/jmxtrans/jmxtrans/pull/288%23issuecomment-148390496%22%2C%20%22https%3A//github.com/jmxtrans/jmxtrans/pull/288%23issuecomment-150071844%22%2C%20%22https%3A//github.com/jmxtrans/jmxtrans/pull/288%23issuecomment-150081793%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/jmxtrans/jmxtrans/pull/288%23issuecomment-100439696%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Thanks%20a%20lot%20for%20bugging%20me%20about%20this.%20I%27m%20working%20on%20changing%20the%20layout%20of%20the%20project%2C%20splitting%20it%20in%20multiple%20modules.%20I%27ll%20apply%20this%20afterward%20to%20make%20sure%20it%20is%20not%20lost%20in%20the%20refactoring.%20Thanks%20for%20your%20patience%20%21%22%2C%20%22created_at%22%3A%20%222015-05-09T07%3A30%3A31Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1415765%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/gehel%22%7D%7D%2C%20%7B%22body%22%3A%20%22Are%20we%20merging%20this%20pull%20request%20sometime%20soon%3F%20As%20the%20JMXTrans%20package%20built%20from%20master%20branch%20is%20ignoring%20the%20%5C%22log%20level%5C%22%20settings%20in%20wrapper.conf.%20JMXTrans%20package%20prints%20debug%20messages%20even%20though%20the%20log%20level%20is%20set%20to%20INFO.%20%22%2C%20%22created_at%22%3A%20%222015-07-13T03%3A37%3A25Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1019609%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/utkarshcmu%22%7D%7D%2C%20%7B%22body%22%3A%20%22This%20should%20actually%20be%20rebased%20as%20the%20structure%20of%20the%20project%20has%20changed%20quite%20a%20bit%20since%20the%20PR%20was%20created.%20I%27ll%20try%20to%20address%20this%20as%20soon%20as%20I%20can%2C%20but%20I%27m%20fairly%20busy%20at%20the%20moment.%20As%20always%2C%20help%20is%20welcomed%20%21%22%2C%20%22created_at%22%3A%20%222015-07-13T11%3A44%3A17Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1415765%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/gehel%22%7D%7D%2C%20%7B%22body%22%3A%20%22Is%20anyone%20actively%20working%20on%20that%3F%22%2C%20%22created_at%22%3A%20%222015-10-13T18%3A53%3A26Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3083638%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/bm371613%22%7D%7D%2C%20%7B%22body%22%3A%20%22Not%20that%20I%20know.%20Reworking%20this%20PR%20should%20be%20fairly%20straigth%20forward.%20Going%20back%20to%20a%20stable%20state%20requires%20a%20bit%20more%20work%3A%5Cr%5Cn%5Cr%5Cn%2A%20deprecating%20%28and%20removing%29%20the%20log4j%20writers%5Cr%5Cn%2A%20ensure%20that%20log4j%20is%20not%20included%20anywhere%5Cr%5Cn%2A%20nice%20to%20have%3A%20provide%20an%20slf4j%20writer%20to%20replace%20the%20log4j%20writer%20%28I%20have%20a%20few%20ideas%20of%20how%20I%20would%20like%20this%20writer%20to%20look%2C%20but%20no%20time%20to%20implement%20it...%29%22%2C%20%22created_at%22%3A%20%222015-10-13T20%3A23%3A54Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1415765%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/gehel%22%7D%7D%2C%20%7B%22body%22%3A%20%22Working%20on%20it.%20PR%20coming%20very%20soon.%20%22%2C%20%22created_at%22%3A%20%222015-10-15T13%3A41%3A25Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1019609%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/utkarshcmu%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40gehel%20%20%5Cr%5CnSo%2C%20should%20I%20just%20deprecate%20jmxtrans-output-log4j%20writer%20and%20create%20jmxtrans-output-slf4j%20writer%3F%20%22%2C%20%22created_at%22%3A%20%222015-10-22T01%3A34%3A06Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1019609%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/utkarshcmu%22%7D%7D%2C%20%7B%22body%22%3A%20%22What%20I%20would%20like%3A%5Cr%5Cn%5Cr%5Cn%2A%20create%20a%20slf4j%20writer%5Cr%5Cn%2A%20replace%20implementation%20if%20log4j%20writer%20by%20a%20simple%20message%20pointing%20the%20user%20to%20the%20new%20writer%5Cr%5Cn%2A%20remove%20all%20references%20to%20log4j%5Cr%5Cn%5Cr%5CnThe%20idea%20I%20have%20for%20the%20slf4j%20writer%20would%20be%20a%20minimal%20writer%20with%20the%20actual%20configuration%20done%20by%20the%20user%20directly%20in%20logback.xml.%20Which%20of%20course%20requires%20some%20documentation%20/%20examples.%5Cr%5Cn%5Cr%5CnBut%20feel%20free%20to%20propose%20something%20else%20if%20you%20have%20a%20good%20idea%21%22%2C%20%22created_at%22%3A%20%222015-10-22T02%3A06%3A43Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1415765%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/gehel%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20b5423bfb924410b7e53f4f3dce1091ae6026d2bd%20jar.properties%206%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/jmxtrans/jmxtrans/pull/288%23discussion_r29990731%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22jar.properties%20is%20actually%20not%20used%20anymore.%20Versionning%20of%20dependencies%20is%20done%20only%20in%20pom.xml.%20I%20should%20cleanup%20that%20jar.properties%20file.%22%2C%20%22created_at%22%3A%20%222015-05-09T07%3A35%3A33Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1415765%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/gehel%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20jar.properties%3AL30-38%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/jmxtrans/jmxtrans/pull/288#issuecomment-100439696'>General Comment</a></b>
- <a href='https://github.com/gehel'></a> Thanks a lot for bugging me about this. I'm working on changing the layout of the project, splitting it in multiple modules. I'll apply this afterward to make sure it is not lost in the refactoring. Thanks for your patience !
- <a href='https://github.com/utkarshcmu'></a> Are we merging this pull request sometime soon? As the JMXTrans package built from master branch is ignoring the "log level" settings in wrapper.conf. JMXTrans package prints debug messages even though the log level is set to INFO.
- <a href='https://github.com/gehel'></a> This should actually be rebased as the structure of the project has changed quite a bit since the PR was created. I'll try to address this as soon as I can, but I'm fairly busy at the moment. As always, help is welcomed !
- <a href='https://github.com/bm371613'></a> Is anyone actively working on that?
- <a href='https://github.com/gehel'></a> Not that I know. Reworking this PR should be fairly straigth forward. Going back to a stable state requires a bit more work:
* deprecating (and removing) the log4j writers
* ensure that log4j is not included anywhere
* nice to have: provide an slf4j writer to replace the log4j writer (I have a few ideas of how I would like this writer to look, but no time to implement it...)
- <a href='https://github.com/utkarshcmu'></a> Working on it. PR coming very soon.
- <a href='https://github.com/utkarshcmu'></a> @gehel
So, should I just deprecate jmxtrans-output-log4j writer and create jmxtrans-output-slf4j writer?
- <a href='https://github.com/gehel'></a> What I would like:
* create a slf4j writer
* replace implementation if log4j writer by a simple message pointing the user to the new writer
* remove all references to log4j
The idea I have for the slf4j writer would be a minimal writer with the actual configuration done by the user directly in logback.xml. Which of course requires some documentation / examples.
But feel free to propose something else if you have a good idea!
- [ ] <a href='#crh-comment-Pull b5423bfb924410b7e53f4f3dce1091ae6026d2bd jar.properties 6'></a> &nbsp;<b><a href='https://github.com/jmxtrans/jmxtrans/pull/288#discussion_r29990731'>File: jar.properties:L30-38</a></b>
- <a href='https://github.com/gehel'></a> jar.properties is actually not used anymore. Versionning of dependencies is done only in pom.xml. I should cleanup that jar.properties file.

<a href='https://www.codereviewhub.com/jmxtrans/jmxtrans/pull/288?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/jmxtrans/jmxtrans/pull/288?approve=1'></a>&nbsp;<a href='https://github.com/jmxtrans/jmxtrans/pull/288'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by gehel** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/jmxtrans/jmxtrans/pull/288'>Refresh</a>.<a href='#crh-update-none'></a>

**Comment by gehel** [comment]
Thanks a lot for bugging me about this. I'm working on changing the layout of the project, splitting it in multiple modules. I'll apply this afterward to make sure it is not lost in the refactoring. Thanks for your patience !

File: jar.properties [hunk 1]
```diff
@@ -30,8 +30,8 @@ slf4j-api.jar=lib/slf4j-api-1.7.6.jar
 slf4j-api.src=lib/slf4j-api-1.7.6-sources.jar
 slf4j-log4j.jar=lib/slf4j-log4j12-1.7.6.jar
 slf4j-log4j.src=lib/slf4j-log4j12-1.7.6-sources.jar
-logback-classic.jar=lib/logback-classic-1.1.1.jar
-logback-core.jar=lib/logback-core-1.1.1.jar
+logback-classic.jar=lib/logback-classic-1.1.2.jar
```
Comment (gehel) [inline jar.properties]: jar.properties is actually not used anymore. Versionning of dependencies is done only in pom.xml. I should cleanup that jar.properties file.

**Comment by utkarshcmu** [comment]
Are we merging this pull request sometime soon? As the JMXTrans package built from master branch is ignoring the "log level" settings in wrapper.conf. JMXTrans package prints debug messages even though the log level is set to INFO.

**Comment by gehel** [comment]
This should actually be rebased as the structure of the project has changed quite a bit since the PR was created. I'll try to address this as soon as I can, but I'm fairly busy at the moment. As always, help is welcomed !

**Comment by bm371613** [comment]
Is anyone actively working on that?

**Comment by gehel** [comment]
Not that I know. Reworking this PR should be fairly straigth forward. Going back to a stable state requires a bit more work:

* deprecating (and removing) the log4j writers
* ensure that log4j is not included anywhere
* nice to have: provide an slf4j writer to replace the log4j writer (I have a few ideas of how I would like this writer to look, but no time to implement it...)

**Comment by utkarshcmu** [comment]
Working on it. PR coming very soon.

**Comment by utkarshcmu** [comment]
@gehel  
So, should I just deprecate jmxtrans-output-log4j writer and create jmxtrans-output-slf4j writer?

**Comment by gehel** [comment]
What I would like:

* create a slf4j writer
* replace implementation if log4j writer by a simple message pointing the user to the new writer
* remove all references to log4j

The idea I have for the slf4j writer would be a minimal writer with the actual configuration done by the user directly in logback.xml. Which of course requires some documentation / examples.

But feel free to propose something else if you have a good idea!

**Comment by gehel** [comment]
replaced by #359

**Pull request closed without merging**